lib/processed-listener.js

```diff
@@ -0,0 +1,81 @@
+'use strict';
+
+const {
+	EventListener
+} = require('@janiscommerce/event-listener');
+
+const Mail = require('@janiscommerce/mail');
+
+const logger = require('lllog')();
+
+const ModelExport = require('./model-export');
+
+const MailMsCallError = 4;
+
+class ProcessedListener extends EventListener {
+
+	get exportModel() {
+		/* istanbul ignore else */
+		if(!this._exportModel)
+			this._exportModel = this.session.getSessionInstance(ModelExport);
+
+		return this._exportModel;
+	}
+
+	get mailService() {
+		/* istanbul ignore else */
+		if(!this._mailService)
+			this._mailService = this.session.getSessionInstance(Mail);
+
+		return this._mailService;
+	}
+
+	get mustHaveClient() {
+		return true;
+	}
+
+	get mustHaveId() {
+		return true;
+	}
+
+	async process() {
+
+		const exportDocument = await this.exportModel.getById(this.eventId);
+
+		if(!exportDocument) {
```

Comment:
```suggestion
		if(!exportDocument || !exportDocument.files) {
			logger.info(`No document or files to export with ID: ${this.eventId} was found.`);
			return;
		}
```
Se pueden unificar los dos `if`